developer_guides/uuid/index.rst

```diff
@@ -0,0 +1,133 @@
+.. _uuid:
+
+UUID Usage in SOF
+#################
+
+Why UUID Needed
+***************
+
+To develope a new audio signal processing component, we traditionally
+implement the component driver with a new unique component type
+introduced, the painpoints with this method including how to keep ABI
+aligned and backward compatible for different version combination of
+topology/driver/firmware for each new added components, there could be
+component type collision if versions are not aligned.
+
+UUIDs (Universally Unique Identifiers) provide a more scalable and
+collision-free way of component identification. UUIDs are currently
+used as the standard interface by all users of the SOF firmware,
+including the tracing subsystem, the topology .m4 files, and the Linux
+topology driver.
+
+Allocating a new UUID for the new added component is recommended, as
+the component type might be replaced by UUID in the future. For the
```

Comment:
so that's unclear now? Is the type replaced yes or no? The previous paragraph strongly hinted that UUID was the only identification method, the use of 'might' makes it a mere possibility. What is the direction?

Reply:
um, per my understanding, the component type will be moved out of the IPC structures, but it is still used inside the firmware, let me re-organize the sentence here.